File: tests/Web.AcceptanceTests/Pages/DashboardPage.cs
namespace LiftAndShift.Web.AcceptanceTests.Pages;

public class DashboardPage(IPage page) : BasePage(page)
{
    public override string PagePath => $"{BaseUrl}/dashboard";

    public Task AssertHeading() => Assertions.Expect(Page.Locator("h1")).ToHaveTextAsync("Dashboard");

    public Task AssertPersonalRecordsSectionVisible() =>
        Assertions.Expect(Page.Locator("h2:has-text('Personal Records')")).ToBeVisibleAsync();

    public Task AssertProgrammeCardVisible() =>
        Assertions.Expect(Page.Locator("a[href='/programme']").First).ToBeVisibleAsync();

    public Task ClickProgrammeLink() => Page.Locator("a[href='/programme']").First.ClickAsync();

    public Task AssertPersonalRecordVisible(string exerciseName) =>
        Assertions.Expect(Page.Locator("table tbody tr", new() { HasText = exerciseName })).ToBeVisibleAsync();
}

File: src/Web/ClientApp/e2e/pages/dashboard.page.ts
import { expect } from '@playwright/test';
import { BasePage, baseUrl } from './base-page';

export class DashboardPage extends BasePage {
  get pagePath(): string {
    return `${baseUrl()}/dashboard`;
  }

  assertHeading() {
    return expect(this.page.locator('h1')).toHaveText('Dashboard');
  }

  assertPersonalRecordsSectionVisible() {
    return expect(this.page.locator("h2:has-text('Personal Records')")).toBeVisible();
  }

  assertProgrammeCardVisible() {
    return expect(this.page.locator("a[href='/programme']").first()).toBeVisible();
  }

  clickProgrammeLink() {
    return this.page.locator("a[href='/programme']").first().click();
  }

  assertPersonalRecordVisible(exerciseName: string) {
    return expect(
      this.page.locator('table tbody tr', { hasText: exerciseName })
    ).toBeVisible();
  }
}
